Ci: Added  Makeformat Command in Ci.yml

Balaji028 opened the pull request:
#58 , Added `make format` as Run command in Ci.yml file .

sh3pik commented on .github/workflows/ci.yml:
```diff
@@ -61,6 +61,7 @@ jobs:
       - if: matrix.driver == 'playwright'
         run: poetry run playwright install chromium
       - run: Xvfb ${{ env.DISPLAY }} -screen 0 1600x1200x24 &
+      - run: make format #This runs all the Linters as Mentioned in Makefile
```
@Balaji028, `make format` will reformat the code "locally." This means that the CI will not fail, and the files will be silently updated. 

However, we want to ensure we commit only the formatted code. One idea for archiving it is to have a separate make command, e.g., `make check-format`, that would fail on each linter instead of reformatting the code. If possible, you can try providing a custom error message on the failure, like "please run make format on your changes!" or something.
WDYT?

Balaji028 replied:
@sh3pik  , Thanks for suggesting  in detail . I will add `make check-format` for each of the linter and provide a custom error message on failure in CI.yml . I will add check-format additional command in existing Makefile in parent directory and in CI  will make use of it using `${{success() }}` / `${{failure() }}` and depending on that we will provide the message which you have suggested . Could you please verify if the approach is fine or if it needs to be changed can you please suggest ?

cgoldberg replied:
The new `Makefile` entry should probably be:
```
check-format:
	poetry run autoflake --check-diff .
	poetry run black --check --diff .
	poetry run isort --check .
	poetry run pyprojectsort --diff
```

This will return a non-zero status if any linter fails, and also output diffs of violations.

Balaji028 replied:
working on that , once done will inform on this .

p0deje replied:
Also, we don't need to run it for all combinations of drivers and models - let's better have a separate job in ci.yml which runs just once.

cgoldberg commented:
@Balaji028 flake8 was added as a linter today, so remember to include that in the `check-format` target.

sudoCommit commented:
Hi, you could use Pre-commit hooks, so that all formats are checked before pushing code.

Or Use both!
- Pre-commit hooks: catch and fix before committing.
- CI format check: enforce formatting in pull requests (in case someone didn't install pre-commit locally).

Balaji028 commented:
Thanks @cgoldberg , @sh3pik  , @p0deje  for Suggesting and helping me . These are the changes I have made .

1) Added Check-format in `Makefile `with `autoflake `, `black `, `isort `, `pyprojectsort`, `flake8 `linters .
2) Added new group for Formatters in pyproject.toml file named as `[tools.poetry.group.format.dependencies]` which includes all the above ones .
3) in `CI.yml`  added a new Job named `Format Checker `  , while installing poetry linter dependencies I haven't specified venv as specified in test job .

Please review this and Suggest me where I can do Improvements , so that I can make better .

sh3pik commented on Makefile:
```diff
@@ -10,6 +10,13 @@ test:
+	poetry run autoflake --check-diff .
+	poetry run black --check --diff .
+	poetry run isort --check .
+	poetry run flake8 .
```
Let's keep the same as in `format` command:
```suggestion
	poetry run flake8 alumnium examples
```
Otherwise, it'll validate the virtual environment folder as well

Balaji028 replied:
Thanks for suggesting , amended the same

sh3pik commented on .github/workflows/ci.yml:
```diff
@@ -25,7 +25,24 @@ env:
+      - run: poetry install --only format
+      - run: make check-format
+      - if: failure()
+        run: echo "please run make format on your changes!"
```
```suggestion
        run: echo "Please run 'make format' on your changes!"
```

Balaji028 replied:
Thanks for suggesting ! amended the same

sh3pik reviewed (commented):
@Balaji028, thanks for the changes! Left two comments. Other than that, it looks good to me!

sh3pik commented:
@sudoCommit, thanks for the idea of pre-commit hooks! I'll add it to the issues list 👍

Balaji028 commented:
Thanks @sh3pik  for reviewing and suggesting the changes ,  Once can you Please review .

cgoldberg reviewed (changes requested):
- The new dependency group in `pyproject.toml` needs to add `flake8-pyproject`.
- To match the other job names in `ci.yml`, `Formatchecker` shouldn't be capitalized (so change it to `formatchecker` or `check-format`).
- The job should have a `name` field.
- To be really pedantic, the `check-format` target should come before `format` in `Makefile` to keep things alphabetical... and the lint jobs within `check-format` should be in alphabetical order to match the `format` target.

cgoldberg reviewed (approved):
LGTM

cgoldberg commented:
@Balaji028 I have a PR open (#69) that edits the `flake8` command in the `Makefile`.

It adds the `tests` directory:

`poetry run flake8 alumnium examples tests`

Can you make sure this is changed in your new `check-format` target as well?

Balaji028 commented:
Hi @sh3pik  , I see `Code Check Style` CI job  is failing because of unable to find the` . tools-version `  file , I am trying to replicate that issue .

Balaji028 commented:
The issue , is because of extra 's' being appended in `.tool-versions` in `ci.yml`  , because of this CI job is failing . sorry for this , have amended now
